src/backends/torch/torchmodule.h

```diff
@@ -42,6 +47,8 @@ namespace dd
     TorchModule();
 
     c10::IValue forward(std::vector<c10::IValue> source);
```

Comment:
How do we choose the GPU in this forward method ? Is it possible to have a single forward and pass the id of one or more GPUs ?

Reply:
I put the multi-gpu stuff in a different method since the calls to torch API are not the same (simple `forward` for single gpu vs call to `data_parallel` for multiple gpus). Usually, the module is located on one device, and this device is selected beforehand. With more GPUs, the model must be cloned dynamically during the forward call (or at least this is the way torch does it).

I can make a single forward and passing the ids

Reply:
I changed `forward_multigpu` to `forward_on_devices` that selects the single or multi gpu mode automatically.